pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -61,13 +63,41 @@ makeWrapper() {
             command="${params[$((n + 1))]}"
             n=$((n + 1))
             echo "$command" >> "$wrapper"
-        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") ]]; then
+        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") || ("$p" == "--suffix-dedup") || ("$p" == "--prefix-dedup") ]]; then
             varName="${params[$((n + 1))]}"
             separator="${params[$((n + 2))]}"
             value="${params[$((n + 3))]}"
             n=$((n + 3))
             if test -n "$value"; then
-                if test "$p" = "--suffix"; then
+                if [[ "$p" == *'-dedup' ]]; then
+                    {
+                        # This abomination removes all occurences of the value that is to be prepended/appended.
+                        # We add (or re-add) the value later. This ensures it's always the last or first value.
+                        # As an example, adding /bin to a colon-separated environment variable would remove:
+                        # - /bin:*
+                        # - *:/bin:*
+                        # - *:/bin
+                        echo "OLDIFS=\$IFS"
+                        echo "value=${value@Q}"
+                        echo "IFS=${separator@Q}"
+                        echo "for v in \$value; do"
```

Comment:
Can't you unroll this loop at compile time? I.e. emit a line for each value, instead of looping at runtime.

Reply:
I've added a commit that does this now, but I'm not sure if it's an improvement. It's probably a bit harder to read and the performance difference is small, at least if you append a realistic amount of items in the wrapper.